cmd/storage.go

```diff
@@ -37,17 +38,22 @@ var storageCreateCmd = &cobra.Command{
 	Example: `  # Create storage of size 1Gb to a component
   odo storage create mystorage --path=/opt/app-root/src/storage/ --size=1Gi
 	`,
-	Args: cobra.ExactArgs(1),
+	Args: cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		client := getOcClient()
 		applicationName, err := application.GetCurrent(client)
 		checkError(err, "")
 		projectName := project.GetCurrent(client)
 		componentName := getComponent(client, storageComponent, applicationName, projectName)
-
-		_, err = storage.Create(client, args[0], storageSize, storagePath, componentName, applicationName)
+		var storageName string
+		if len(args) != 0 {
+			storageName = args[0]
+		} else {
+			storageName = componentName + "-" + util.GenerateRandomString(4)
```

Comment:
@surajnarwade This LGTM. Is this the same way that OpenShift / Kubernetes is generating their random strings for their pods? We should replicate / make this similar.

Reply:
Kubernetes, OpenShift way is bit different, Since we already had `GenerateRandomString`. I reused it :)